index.js

```diff
@@ -4,140 +4,219 @@ window.addEventListener("load", () => {
   const eqButton = document.querySelector(".eButton");
   const numDisplay = document.querySelector(".numDisplay");
   const opArray = ["add", "subtract", "multiply", "divide"];
-  let currentString = numDisplay.textContent;
+  let currentString = "";
+  //This should always have STRING data. numDisplay, despite the name, should also always have STRING data.
   let firstNum = "";
+  //This should always have NUMBER data, OR a BLANK STRING (bad practice - figure out a better solution later).
   let currentOp = "";
+  //This is STRING.
+  let clearState = "ac";
+  //This is STRING.
   let errLock = false;
+  //This is BOOLEAN.
+
+  function clearCheck() {
+    //There's gotta be a more elegant way of doing this, right?
+    if (currentString != "") {
+      clearState = "c";
+      opButtons[5].textContent = clearState;
+    } else {
+      clearState = "ac";
+      opButtons[5].textContent = clearState;
+    }
+  }
+
+  function rounder(num) {
+    //This function accepts NUMBER data and returns STRING data.
+    if (Math.abs(num) > 999999999 || Math.abs(num) < 0.00000001) {
+      return num.toExponential(5);
+    } else {
+      let n = 10 - num.toString().split(".")[0].match(/\d/g).length;
+      //line 16 reads as follows: "Subtract the number of digits to the left of the decimal point from 10"
+      return parseFloat(num.toFixed(n)).toString();
+    }
+  }
+
+  function equation() {
+    if (currentOp == "add") {
+      currentString = firstNum + Number(currentString);
```

Comment:
Here and in the other operations, you're assigning the result of the calculation to currentString, which is supposed to be the text typed by the user. You're resetting it to an empty string a few lines later, so this doesn't currently introduce any bugs, but for good programming practice you should be using a new temporary variable like `const result = ` instead of using an existing variable.

Reply:
Updated. Created variable 'solution' to temporarily store result of operations instead of using 'currentString = currentString + '...